### DISCUSSION TITLE
请教umi4中怎么配置支持es2024的新特性？

### DISCUSSION THREAD

**Discussion by leslie555** [umijs/umi#12266] [Q&A] [answered]
想要支持Object.groupBy等新方法。

**Comment by fz6m** [comment]
早就有了吧，core-js 会提前支持 proposal 的方法，看下你的 `src/.umi/core/polyfill.ts` ，里面有所有 polyfill 的列表，其中 `core-js/modules/es.object.group-by.js` 就是 `Object.groupBy` ，我看最新 umi 4 版本是有的，直接用就行了。

**Reply by leslie555** [reply]
确实是可以的， 只是ts目前还没有在Object中添加groupBy属性

**Reply by fz6m** [reply, marked as answer]
[TS 5.4](https://devblogs.microsoft.com/typescript/announcing-typescript-5-4/#objectgroupby-and-mapgroupby) 就在内置类型里加入这部分了，你把项目里的 TS 依赖升级到 5.4 然后配置 vscode 采用项目的 TS 引擎即可（或者升级 vscode 版本到最新，从而自动采用 vscode 内置 TS 版本，或者使用 TS nightly ）。
